packages/contracts-bedrock/test/kontrol/kontrol/scripts/make-summary-deployment.sh

```diff
@@ -0,0 +1,33 @@
+# create deployments/hardhat/.deploy and snapshots/state-diff/Deploy.json if necessary
+if [ ! -d "deployments/hardhat" ]; then
+  mkdir deployments/hardhat;
+fi
+if [ ! -f "deployments/hardhat/.deploy" ]; then
+  touch deployments/hardhat/.deploy;
+fi
+if [ ! -d "snapshots/state-diff" ]; then
+  mkdir snapshots/state-diff;
+fi
+if [ ! -f "snapshots/state-diff/Deploy.json" ]; then
+  touch snapshots/state-diff/Deploy.json;
+fi
+
+# replace mustGetAddress by getAddress in Deploy.s.sol
+sed -i 's/mustGetAddress/getAddress/g' ./scripts/Deploy.s.sol
+
```

Comment:
`sed` behaves differently on macOS and linux, so when I run this it errors with:
```
sed: 1: "./scripts/Deploy.s.sol": invalid command code .
```

We'll have to replace with `awk`, this should work:

```suggestion
awk '{gsub(/mustGetAddress/, "getAddress")}1' ./scripts/Deploy.s.sol > temp && mv temp ./scripts/Deploy.s.sol
```

Additionally, let's make sure to undo the replacements at the end of the script (and even if it fails). Maybe easiest way is to reset that file with git in a trap?

Reply:
Ouch, I didn't know about that `sed` discrepancy! Will replace it with the `awk` command.
It seems reasonable to undo the replacements. I'm unfamiliar with the approach you're suggesting. Could you elaborate on it? Thank you!

Reply:
> It seems reasonable to undo the replacements. I'm unfamiliar with the approach you're suggesting. Could you elaborate on it? Thank you!

I was thinking of using `git reset` to revert local changes to match the origin, but that would also reset local changes, so you're better off just creating then restoring a backup, like this:

```sh
DEPLOY_SCRIPT="./scripts/Deploy.s.sol"

# Create a backup
cp "$DEPLOY_SCRIPT" "$DEPLOY_SCRIPT.bak"

# Modify the file, do any other operations here
# --- snip ---

# Restore the file from the backup
cp "$DEPLOY_SCRIPT.bak" "$DEPLOY_SCRIPT"
rm "$DEPLOY_SCRIPT.bak"
```